### ISSUE TITLE
Assess exploding gradients.

### ISSUE THREAD

**Issue by lar9482** [lar9482/MNIST#1]
I've done a little experimentation with batch training on the MNIST dataset.
It seems like exploding gradients are a major problem right now.

Gradient clipping and/or weight regularization(L1) needs to be looked into.

**Comment by lar9482** [comment]
Implemented softmax activation function with in combination with the cross entropy function.
This seems to fix the problem for now.
